src/hooks/useOperatorTrades.ts

```diff
@@ -1,68 +1,44 @@
-import { Trade as TradeMetaData } from '@cowprotocol/cow-sdk'
-import { getTrades, Order, Trade } from 'api/operator'
+import { getTrades, Order, RawTrade, Trade } from 'api/operator'
 import { useCallback, useEffect, useState } from 'react'
 import { useNetworkId } from 'state/network'
 import { Network, UiError } from 'types'
 import { transformTrade } from 'utils'
-
-type Params = {
-  owner?: string
-  orderId?: string
-}
+import { web3 } from 'apps/explorer/api'
 
 type Result = {
   trades: Trade[]
   error?: UiError
   isLoading: boolean
 }
 
-/**
- * Fetches trades for given filters
- * When no filter is given, fetches all trades for current network
- */
-export function useTrades(params: Params): Result {
-  const { owner, orderId } = params
+type TradesTimestamps = { [txHash: string]: number }
 
-  const [isLoading, setIsLoading] = useState(false)
-  const [error, setError] = useState<UiError>()
-  const [trades, setTrades] = useState<Trade[]>([])
+const tradesTimestampsCache: { [blockNumber: number]: number } = {}
```

Comment:
Are we caching forever?

Reply:
I don't care a lot, since a refresh solves it...

Reply:
Timestamp of a block is constant, so it's ok

Reply:
yes, my point was more like we never garbage collect anything, but as i said above, doesn't matter too much, a refresh will make our app forget about the cache

😈 Making a bit of trouble:
> Timestamp of a block is constant, so it's ok

This is not true necessarily (reorgs). We can ignore this anyways, I'm not concerned with this at all